crates/api_models/src/feature_matrix.rs

```diff
@@ -0,0 +1,51 @@
+use std::collections::HashSet;
+
+use serde::{Deserialize, Serialize};
+use utoipa::ToSchema;
+
+use crate::enums::{
+    CaptureMethod, Connector, CountryAlpha2, Currency, EventClass, PaymentConnectorCategory,
+    PaymentMethod, PaymentMethodType,
+};
+
+#[derive(Default, Debug, Deserialize, Serialize, Clone, ToSchema)]
+pub struct FeatureMatrixRequest {
+    // List of connectors for which the feature matrix is requested
+    pub connectors: Option<Vec<Connector>>,
+}
+
+#[derive(Debug, ToSchema, Serialize)]
+pub struct SupportedPaymentMethod {
+    pub payment_method: PaymentMethodType,
+    pub supports_mandate: bool,
+    pub supports_refund: bool,
+    pub supported_capture_methods: Vec<CaptureMethod>,
+    pub supported_countries: Option<HashSet<CountryAlpha2>>,
+    pub supported_currencies: Option<HashSet<Currency>>,
+}
+
+#[derive(Debug, ToSchema, Serialize)]
+pub struct SupportedPaymentMethodTypes {
+    pub payment_method_type: PaymentMethod,
```

Comment:
Shouldn't this be payment method and the underlying subtype should be `payment_method_type` right